File: library/src/main/java/me/toptas/rssconverter/RssConverterFactory.java
package me.toptas.rssconverter;

import java.lang.annotation.Annotation;
import java.lang.reflect.Type;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import retrofit2.Converter;
import retrofit2.Retrofit;

/**
 * A {@link Converter.Factory converter} which uses {@link XMLParser} to parse RSS feeds.
 */

public final class RssConverterFactory extends Converter.Factory {

    /**
     * Creates an instance
     *
     * @return instance
     */
    public static RssConverterFactory create() {
        return new RssConverterFactory();
    }

    /**
     * Constructor
     */
    private RssConverterFactory() {
        super();
    }

    @Override
    public Converter<ResponseBody, ?> responseBodyConverter(Type type, Annotation[] annotations, Retrofit retrofit) {
        return new RssResponseBodyConverter<>();
    }

    @Override
    public Converter<?, RequestBody> requestBodyConverter(Type type, Annotation[] parameterAnnotations, Annotation[] methodAnnotations, Retrofit retrofit) {
        return super.requestBodyConverter(type, parameterAnnotations, methodAnnotations, retrofit);
    }

    @Override
    public Converter<?, String> stringConverter(Type type, Annotation[] annotations, Retrofit retrofit) {
        return super.stringConverter(type, annotations, retrofit);
    }
}

File: library/src/main/java/me/toptas/rssconverter/RssConverterFactory.kt
package me.toptas.rssconverter

import java.lang.reflect.Type

import okhttp3.ResponseBody
import retrofit2.Converter
import retrofit2.Retrofit

/**
 * A [converter][Converter.Factory] which uses [XMLParser] to parse RSS feeds.
 */

class RssConverterFactory
/**
 * Constructor
 */
private constructor() : Converter.Factory() {

    override fun responseBodyConverter(type: Type?,
                                       annotations: Array<Annotation>?,
                                       retrofit: Retrofit?):
            Converter<ResponseBody, *> = RssResponseBodyConverter()

    companion object {

        /**
         * Creates an instance
         *
         * @return instance
         */
        fun create(): RssConverterFactory {
            return RssConverterFactory()
        }
    }
}
